src/com/elikill58/negativity/spigot/packets/PacketType.java

```diff
@@ -0,0 +1,296 @@
+package com.elikill58.negativity.spigot.packets;
+
+import com.elikill58.negativity.spigot.SpigotNegativity;
+
+public interface PacketType {
+
+	public String name();
+	public String getPacketName();
+	public String getFullName();
+	
+	static boolean LOG_UNKNOW_PACKET = false;
+	static String CLIENT_PREFIX = "PacketPlayIn", SERVER_PREFIX = "PacketPlayOut", LOGIN_PREFIX = "PacketLogin", STATUS_PREFIX = "PacketStatus";
+	
+	public static PacketType getType(String packetName) {
+		if(packetName.startsWith(CLIENT_PREFIX)) {
+			for(Client client : Client.values())
+				if(client.getFullName().equalsIgnoreCase(packetName))
+					return client;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow client packet " + packetName);
+			return Client.UNSET;
+		} else if(packetName.startsWith(SERVER_PREFIX)) {
+			for(Server srv : Server.values())
+				if(srv.getFullName().equalsIgnoreCase(packetName))
+					return srv;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow server packet " + packetName);
+			return Server.UNSET;
+		} else if(packetName.startsWith(LOGIN_PREFIX)) {
+			for(Login login : Login.values())
+				if(login.getFullName().equalsIgnoreCase(packetName))
+					return login;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow login packet " + packetName);
+			return Login.UNSET;
+		} else if(packetName.startsWith(STATUS_PREFIX)) {
+			for(Status status : Status.values())
+				if(status.getFullName().equalsIgnoreCase(packetName))
+					return status;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow status packet " + packetName);
+			return Status.UNSET;
+		} else {
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow packet " + packetName);
+			return null;
+		}
+	}
+	
+	public static enum Client implements PacketType {
```

Comment:
```suggestion
	enum Client implements PacketType {
```
Pareil pour les autres enums